### DISCUSSION TITLE
Sciatic nerve toluidine blue image analysis on AxonDeepSeg

### DISCUSSION THREAD

**Discussion by apuer11** [neuropoly/axondeepseg#842] [Help needed]
Hi everyone!
I followed all of the steps to install AxonDeepSeg as a plugin on napari and everything seems to be set up correctly according to the quick check and running the samples for the models. I am just having trouble determining what are the steps I should take to prep my image to be analyzed using the plug in. I have seen some posts mentioning grayscale and cropping the images to a certain size but I am not sure of exactly everything I should be doing for the model to easily run my image. I was also wondering how I would determine the number of axons in the entire cross section if this tool only accepts a small square of the whole image at a time? I am not sure if the contrast of my images are not good enough and that is causing the problem because when I finally got it to run it took a while and did not create the red and blue layers. Thank you for your help in advance!
![Capture](https://github.com/axondeepseg/axondeepseg/assets/142545380/8082b059-3bd7-42e2-a34e-df5bc4cba258)

> and I am prompted for pixel size, I click ok and the program just doesn't segment my axons/myelin.
@SPACEADITYA do you enter a pixel size? Could you look in the terminal (where you launched FSLeyes) to see if there are any error messages there after you try segmenting the image?
_Originally posted by @mathieuboudreau in https://github.com/axondeepseg/axondeepseg/discussions/614#discussioncomment-2368758_

**Comment by mathieuboudreau** [comment]
@apuer11 did you get answers to your questions / did you problem get resolved? Or did you close this issue accidentally?

**Comment by apuer11** [comment]
Thank you for the speedy reply! I did not get my problem resolved yet, this was my first time posting on here and I changed my mind thinking that posting under discussions might be better so I resolved this post so it wasn't being asked in two locations.

**Comment by Stoyan-I-A** [comment]
@apuer11 Thank you for reaching out. I have a few questions regarding your image.
1st- Is your image _bit depth_ 8-bits or 16-bits? You can find this information by right-clicking your image file, then go to `properties-> Details -> Image`

The package we use for the segmentation currently has a problem with 16 bit images. So if you have a 16-bit image, you'll have to convert your image to a 8-bit one. (Check https://www.youtube.com/watch?v=exLmtLBrgdY)

2nd- If you do have 8-bit image, can you copy and paste here the output of the terminal?

**Comment by apuer11** [comment]
@Stoyan-I-A It says 24 for the bit depth of the normal image, but after changing it to 8 bit on Image J and putting a cropped square in napari since it does not accept the full image the output is the following:
![IMG_4937](https://github.com/axondeepseg/axondeepseg/assets/142545380/7defe004-4898-4d11-ae65-e31b48683881)

**Comment by Stoyan-I-A** [comment]
Hmmm the result looks very similar to what this user here had: https://github.com/axondeepseg/axondeepseg/discussions/736 . And in that case, the problem was that he/she had 16 bit images.

You could try to segment your image with different models and zoom factors, but I would still expect better results if the section you have is indeed a 8-bit image.

Do you mind sharing a section of your image? Like the cropped section you used on your latest reply.

Do you know the pixel size as well?

**Comment by apuer11** [comment]
So sorry for the late reply, I was away from the images and then I was trying to find a simpler way to count these axons using Matlab, but unfortunately the images seem to vary too much in contrast (how good/dark the sample stain is) for all the axons to be counted and their irregular shapes are making diameter measurements very difficult. I am attaching the images you requested below. I appreciate everyone's help, but I do not want to waste anyone's time so do you think this tool will be able to distinguish axons in images that are very light and how long does it take on average to get counts and measurements for each nerve cross-section usually (since the program seems to process one square of the cross-section at a time)? Thank you again!
![Screenshot 2023-09-07 115324](https://github.com/axondeepseg/axondeepseg/assets/142545380/2aeb596b-70b3-4708-a580-876311708a6c)
![Screenshot 2023-09-07 115038](https://github.com/axondeepseg/axondeepseg/assets/142545380/583ec68c-3b0b-43ef-9cb6-9aafbdd74fee)

**Comment by Stoyan-I-A** [comment]
It seems like GitHub's compression reduced this image's dimensions to 943 x 885 px which makes the image blurry when we try to zoom in. Could you send a smaller crop of the image? Perhaps just a 1024 x 1024 px portion of the image.
Another possibility would be to store your image in a cloud service (like Google Drive or OneDrive) and share us the link at axondeepseg@googlegroups.com . 

>  do you think this tool will be able to distinguish axons in images that are very light
 
It depends, it might sometimes struggle, but we can also try to do some preprocesssing to see if we can find a way to increase the contrast using some filters.

>  how long does it take on average to get counts and measurements for each nerve cross-section

It depends on your computer's processing speed. But I expect it to take a few minutes at most. That being said, the count and measurements is not the most challenging part. The challenging part is to get a good segmentation.

**Comment by mathieuboudreau** [comment]
Converting this issue into a user-discussion
